ruby_on_rails/robots_txt.md

```diff
@@ -0,0 +1,40 @@
+# robots.txt
+
+It is time to configure the `robots.txt` file properly, to avoid crawlers to find our develop and testing environments.
+The master environment should be the only one searchable in the end.
+
+Make sure that there is a `robots.txt` file in the public folder or your project (Rails should have created it).
+This file will only be used in environments where the `BLOCK_ROBOTS` environment variable is not set. If it is set then a custom middleware catches calls to `/robots.txt`
+
+Create the following file:
+
+```ruby
+# app/middleware/robots_txt.rb
+
+class RobotsTxt
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    if env['PATH_INFO'] == '/robots.txt' && ENV['BLOCK_ROBOTS'].present?
```

Comment:
I'd leave away the environment check because we only use this middleware if it is.

Reply:
I don't really have a good point for that: I tend to duplicate this check in my projects and have it both in the middleware itself and in the project, but I am happy to leave this away.